pilot/pkg/xds/xdsgen.go

```diff
@@ -187,3 +188,52 @@ func ResourceSize(r model.Resources) int {
 	}
 	return size
 }
+
+func xdsNeedsPush(req *model.PushRequest, proxy *model.Proxy, typeURL string,
+	checkConfigUpdateFunc func(req *model.PushRequest, proxy *model.Proxy) bool,
+) bool {
+	if proxy.Type == model.Ztunnel {
+		// Not supported for ztunnel
+		return false
+	}
+	if req == nil {
+		return true
+	}
+	// Special handling for wildcard type URLs - CDS and LDS.
+	if xds.IsWildcardTypeURL(typeURL) {
```

Comment:
This seems technically correct but only coincidentally. Waypoints need LDS and CDS pushes for Address because Address impacts LDS and CDS -- not because they are wildcars. They just happen to be the same resources.